### PR TITLE
Ckbtc install

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to have e2e tests for ckbtc functionality in the nns-dapp repo.

# Changes
- Add a CKBTC deployment script
  - This is the script from the nns-dapp repo, with variables such as network taken from command line flags.
- Include ckbtc canisters in the stock snapshot

# Tests
* The ckbtc deployment contains a self-test.  Additionally, there is a set of tests for the self-test itself.

### CODE REVIEW

File: bin/dfx-snapshot-stock-make
Comment (bitdivine) [inline bin/dfx-snapshot-stock-make:28]: @lmuntaner - The [ckbtc deployment script in the nns-dapp repo](https://github.com/dfinity/nns-dapp/blob/main/scripts/ckbtc/deploy-ckbtc) deploys with the prefix `ckbtc_test_` but dfx.json has the prefix `ckbtc_` (no test).  Which would you like, or both?

File: bin/dfx-ckbtc-deploy
Comment (bitdivine) [inline bin/dfx-ckbtc-deploy:1]: This file is imported from nns-dapp with the following changes:
- Add the bash cli parser and `--help` message.
- Make the prefix a parameter.

File: bin/dfx-ckbtc-deploy
Comment (bitdivine) [inline bin/dfx-ckbtc-deploy:59]: Should any of this be variable?  Do you need both test ckbtc and main ckbtc and if so, should they have different symbols and token names?

File: bin/dfx-snapshot-stock-make
Comment (bitdivine) [inline bin/dfx-snapshot-stock-make:1]: Here the ckbtc canisters are added to the stock snapshot, so that they are available for nns-dapp e2e tests.

Are there any exit criteria that need to be tested?  I can check that the canisters have been created.  Is anything else needed to classify a ckbtc deployment as fit for nns-dapp testing?

File: bin/dfx-snapshot-stock-make
Comment (lmuntaner) [reply bin/dfx-snapshot-stock-make:28]: no `test`. I thought I removed that to be honest.

Thanks for doing this.

DOn't you need to add them to dfx.json?

Comment (bitdivine) [reply bin/dfx-snapshot-stock-make:28]: `dfx-ckbtc-import` adds entries to dfx.json, if they are not already there.  If there are already entries, they are used.

File: bin/dfx-canister-check-wasm-hash
Comment (bitdivine) [inline bin/dfx-canister-check-wasm-hash:1]: This is imported from the nns-dapp repo.  It probably belongs here, along with the other polyfills, anyway.

File: bin/dfx-ckbtc-deploy
Comment (lmuntaner) [reply bin/dfx-ckbtc-deploy:65]: @peterpeterparker what do you think?

Is one enough for testnet?

**Review by lmuntaner — COMMENTED** [review]
LGTM, but I'll wait for @peterpeterparker 's approval also

File: bin/dfx-ckbtc-deploy
Comment (peterpeterparker) [reply bin/dfx-ckbtc-deploy:65]: Ideally our test environment would deploy ckTESTBTC and - or connect to - a Bitcoin testnet.

ckBTC without the Bitcoin testnet is just another ICRC token as any Sns token.

Comment (bitdivine) [reply bin/dfx-ckbtc-deploy:65]: Sounds good.  My impression is that deploying ckTESTBTC should be possible but would mean more code, so it would be better to do that part in a separate PR.  If we aim to deploy ckTESTBTC, is the data in this PR compatible?  E.g. is the token name fine?  We can always adjust later but if it's right from the start we don't risk breaking e2e tests that use the first token name.

If the code here is consistent with the goal, can we merge please?

**Review by lmuntaner — APPROVED** [review]

File: bin/dfx-ckbtc-deploy
Comment (lmuntaner) [reply bin/dfx-ckbtc-deploy:65]: The effort to have `ckTESTBTC` is big. Even the FI team don't have it. Maybe we should wait until they do?

We have a lot of things in our plates at the moment.

**Pull request merged**
